managegroup.php

```diff
@@ -195,38 +197,42 @@
 
 	redirect("managegroup.php?gid={$gid}", $lang->join_requests_moderated);
 }
-elseif($mybb->input['action'] == "joinrequests")
-{
-	$users = "";
-	$plugins->run_hooks("managegroup_joinrequests_start");
-
-	$query = $db->query("
-		SELECT j.*, u.uid, u.username, u.postnum, u.regdate
-		FROM ".TABLE_PREFIX."joinrequests j
-		LEFT JOIN ".TABLE_PREFIX."users u ON (u.uid=j.uid)
-		WHERE j.gid='{$gid}' AND j.uid != 0
-		ORDER BY u.username ASC
-	");
-	while($user = $db->fetch_array($query))
-	{
-		$user['reason'] = htmlspecialchars_uni($user['reason']);
-		$altbg = alt_trow();
-		$regdate = my_date($mybb->settings['dateformat'], $user['regdate']);
-		$user['username'] = htmlspecialchars_uni($user['username']);
-		$user['profilelink'] = build_profile_link($user['username'], $user['uid']);
-		eval("\$users .= \"".$templates->get("managegroup_joinrequests_request")."\";");
-	}
-	if(!$users)
-	{
-		error($lang->no_requests);
-	}
-	$lang->join_requests = $lang->sprintf($lang->join_requests_title, htmlspecialchars_uni($usergroup['title']));
-
-	$plugins->run_hooks("managegroup_joinrequests_end");
 
-	eval("\$joinrequests = \"".$templates->get("managegroup_joinrequests")."\";");
-	output_page($joinrequests);
+elseif ($mybb->input['action'] == "joinrequests") {
+    $users = [];
+    $plugins->run_hooks("managegroup_joinrequests_start");
+
+    $query = $db->query("
+        SELECT j.*, u.uid, u.username, u.postnum, u.regdate
+        FROM " . TABLE_PREFIX . "joinrequests j
+        LEFT JOIN " . TABLE_PREFIX . "users u ON (u.uid=j.uid)
+        WHERE j.gid='{$gid}' AND j.uid != 0
+        ORDER BY u.username ASC
+    ");
+    while ($user = $db->fetch_array($query)) {
+        $user['reason'] = htmlspecialchars_uni($user['reason']);
```

Comment:
The use of `htmlspecialchars_uni()` here (and perhaps below?) will cause double escaping.